snooty/parser.py

```diff
@@ -560,6 +568,12 @@ def dispatch_departure(self, node: tinydocutils.nodes.Node) -> None:
         elif isinstance(popped, n.Directive) and popped.name == "wayfinding":
             self.handle_wayfinding(popped)
 
+        elif isinstance(popped, n.Directive) and popped.name == "method-selector":
```

Comment:
Makes sense. Happy to factor out an `if not isinstance(popped, n.Directive)` if we want to bundle it as part of this PR. Interested to see what a refactor of this would look like 👀

Reply:
My thinking is instead of God's Ugliest elif chain this should be a list of handlers, ideally each in their own file. Closer to what we do in the postprocessor.